habitat_sim/utils/data/data_structures.py

```diff
@@ -0,0 +1,74 @@
+class LinkedListNode:
```

Comment:
This can be done with an OrderedDict like here: https://github.com/tkem/cachetools/blob/master/cachetools/lru.py

Reply:
That was a lot simpler, didn't know that existed! Fixed now.